File: FindTaskByName/VB/FindTaskByName/FindTaskByName.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.IO
Imports Aspose.Tasks.Util

Namespace Aspose.Tasks.FindTaskByName
	''' <summary>
	''' This example shows how to combine some conditions.
	''' </summary>
	Friend Class FindTaskByName
		''' <summary>
		''' Main method
		''' </summary>
        <STAThread()> _
        Shared Sub Main(ByVal args As String())
            ' Provide path to project file, name of the task you want to search 
            args = New String() {"Path to project file", "Name of the task to find"}
            Try
                If args.Length <> 2 Then
                    System.Console.WriteLine("Usage: Aspose.Tasks.FindByName <input file name> <taskName>")
                Else
                    Find(args(0), args(1))
                End If
            Catch ex As System.Exception
                System.Console.WriteLine(ex.StackTrace)
            End Try
            Console.ReadKey()
        End Sub

		''' <summary>
		''' Finds taks by name and prinst task name, uid, id, date start and date end
		''' </summary>
		''' <param name="fileName"></param>
		''' <param name="taskName"></param>
        Private Shared Sub Find(ByVal fileName As String, ByVal taskName As String)
            ' reads project from file
            Dim reader As ProjectReader = New ProjectReader()
            Dim project As Project = reader.Read(New FileStream(fileName, FileMode.Open))

            ' Searching task
            Dim task As Task = TaskUtils.Find(project.RootTask, New NameCondition(taskName))

            ' Prints task parameters
            If Not task Is Nothing Then
                Console.WriteLine("Name   : " & task.Name)
                Console.WriteLine("UID    : " & task.Uid)
                Console.WriteLine("ID     : " & task.Id)
                Console.WriteLine("Start  : " & task.Start)
                Console.WriteLine("Finish : " & task.Finish)
            End If
        End Sub
	End Class

	''' <summary>
	''' Define find condition. Method Check() of this class returns true if task name starts with the pattern otherwise returns false;
	''' </summary>
	Friend Class NameCondition
		Implements ICondition
		Private pattern As String

		Public Sub New(ByVal pattern As String)
			Me.pattern = pattern
		End Sub

		Public Function Check(ByVal el As Object) As Boolean Implements ICondition.Check
			Dim task As Task = CType(el, Task)
			If Not task.Name Is Nothing Then
				Return task.Name.StartsWith(pattern)
			End If
			Return False
		End Function
	End Class
End Namespace

File: FindTaskByName/C#/FindTaskByName/FindTaskByName.cs
using System;
using System.IO;
using Aspose.Tasks.Util;

namespace Aspose.Tasks.FindTaskByName
{
    /// <summary>
    /// This example shows how to combine some conditions.
    /// </summary>
    internal class FindTaskByName
    {
        /// <summary>
        /// Main method
        /// </summary>
        [STAThread]
        private static void Main(string[] args)
        {
            // Provide path to project file and name of the task you want to search 
            args = new string[] { @"Path to project file", "Name of the task to find" };
            try
            {
                if (args.Length != 2)
                {
                    System.Console.WriteLine("Usage: Aspose.Tasks.FindByName <input file name> <taskName>");
                }
                else
                {
                    Find(args[0], args[1]);
                }
            }
            catch (System.Exception ex)
            {
                System.Console.WriteLine(ex.StackTrace);
            }
        }

        /// <summary>
        /// Finds taks by name and prinst task name, uid, id, date start and date end
        /// </summary>
        /// <param name="fileName"></param>
        /// <param name="taskName"></param>
        private static void Find(string fileName, string taskName)
        {
            // reads project from file
            ProjectReader reader = new ProjectReader();
            Project project = reader.Read(new FileStream(fileName, FileMode.Open));

            // Searching task
            Task task = TaskUtils.Find(project.RootTask, new NameCondition(taskName));

            // Prints task parameters
            if (task != null)
            {
                Console.WriteLine("Name   : " + task.Name);
                Console.WriteLine("UID    : " + task.Uid);
                Console.WriteLine("ID     : " + task.Id);
                Console.WriteLine("Start  : " + task.Start);
                Console.WriteLine("Finish : " + task.Finish);
            }
            Console.ReadKey();
        }
    }

    /// <summary>
    /// Define find condition. Method Check() of this class returns true if task name starts with the pattern otherwise returns false;
    /// </summary>
    internal class NameCondition : ICondition
    {
        private string pattern;

        public NameCondition(String pattern)
        {
            this.pattern = pattern;
        }

        public bool Check(object el)
        {
            Task task = (Task)el;
            if (task.Name != null)
            {
                return task.Name.StartsWith(pattern);
            }
            return false;
        }
    }
}